File: Online.Store.Core/DTOs/Order.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Online.Store.Core.DTOs
{
    public class Order
    {
        public Order()
        {
            OrderDetails = new List<OrderDetail>();
        }
        public int Id { get; set; }
        public string UserId { get; set; }
        public DateTime DateCreated { get; set; }

        public ICollection<OrderDetail> OrderDetails { get; set; }
    }

    public class OrderDetail
    {
        public int Id { get; set; }
        public int OrderId { get; set; }
        public Order Order { get; set; }
        public string ProductId { get; set; }
        public double Price { get; set; }
        public int Quantity { get; set; }
    }
}

File: Online.Store/ClientApp/app/models/order.ts
export class Order  {
    public id: number;
    public userId: string;
    public dateCreated: Date;
    public orderDetails: OrderDetail[];
}

export class OrderDetail {
    public id: number;
    public productId: string;
    public productTitle: string;
    public productPrice: number;
    public productImage: string;
    public quantity: number;
}